docker/orca-nightly/Dockerfile

```diff
@@ -0,0 +1,146 @@
+# java
+    wget $JDK_URL && \
+    gunzip jdk-$JDK_VERSION-linux-x64.tar.gz && \
+    tar -xf jdk-$JDK_VERSION-linux-x64.tar -C /opt && \
+    rm jdk-$JDK_VERSION-linux-x64.tar && \
+    mv /opt/jdk* /opt/jdk$JDK_VERSION && \
+    ln -s /opt/jdk$JDK_VERSION /opt/jdk && \
+# spark
+    wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop2.7.tgz && \
+    tar -zxvf spark-${SPARK_VERSION}-bin-hadoop2.7.tgz && \
+    mv spark-${SPARK_VERSION}-bin-hadoop2.7 /opt/spark && \
+    rm spark-${SPARK_VERSION}-bin-hadoop2.7.tgz && \
+    cp /opt/spark/kubernetes/dockerfiles/spark/entrypoint.sh /opt && \
+    # remove log4j 1.x jars
+    rm -f ${SPARK_HOME}/jars/log4j-1.2.17.jar && \
+    rm -f ${SPARK_HOME}/jars/slf4j-log4j12-1.7.16.jar && \
+    rm -f ${SPARK_HOME}/jars/apache-log4j-extras-1.2.17.jar && \
+    wget -P ${SPARK_HOME}/jars/ https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-1.2-api/2.17.1/log4j-1.2-api-2.17.1.jar && \
+    wget -P ${SPARK_HOME}/jars/ https://repo1.maven.org/maven2/org/slf4j/slf4j-reload4j/1.7.35/slf4j-reload4j-1.7.35.jar && \
+    wget -P ${SPARK_HOME}/jars/ https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-api/2.17.1/log4j-api-2.17.1.jar && \
+    wget -P ${SPARK_HOME}/jars/ https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-core/2.17.1/log4j-core-2.17.1.jar && \
+    wget -P ${SPARK_HOME}/jars/ https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-slf4j-impl/2.17.1/log4j-slf4j-impl-2.17.1.jar
+
+ADD ./log4j2.xml ${SPARK_HOME}/conf/log4j2.xml
+ADD ./spark-defaults.conf ${SPARK_HOME}/conf/spark-defaults.conf
+RUN ln -fs /bin/bash /bin/sh
+RUN if [ $SPARK_VERSION = "3.1.3" ]; then \
+        rm $SPARK_HOME/jars/okhttp-*.jar && \
+        wget -P $SPARK_HOME/jars https://repo1.maven.org/maven2/com/squareup/okhttp3/okhttp/3.8.0/okhttp-3.8.0.jar; \
+    elif [ $SPARK_VERSION = "2.4.6" ]; then \
+        rm $SPARK_HOME/jars/kubernetes-client-*.jar && \
+        wget -P $SPARK_HOME/jars https://repo1.maven.org/maven2/io/fabric8/kubernetes-client/4.4.2/kubernetes-client-4.4.2.jar; \
+    fi
+
+# stage.2 bigdl
+FROM ubuntu:20.04 as bigdl
+ARG SPARK_VERSION
+ARG BIGDL_VERSION
+ARG DEBIAN_FRONTEND
+ENV SPARK_VERSION               ${SPARK_VERSION}
+ENV BIGDL_VERSION               ${BIGDL_VERSION}
+ENV BIGDL_HOME                  /opt/bigdl-${BIGDL_VERSION}
+
+RUN apt-get update --fix-missing && \
+    apt-get install -y apt-utils vim curl nano wget unzip  git
+ADD ./download-bigdl.sh /opt
+
+RUN chmod a+x /opt/download-bigdl.sh && \
+    mkdir -p /opt/bigdl-examples/python
+RUN /opt/download-bigdl.sh && \
+    rm bigdl*.zip
+
+# stage.3 bigdl-tutorials
+FROM ubuntu:20.04
+ARG BIGDL_VERSION
+ARG SPARK_VERSION
+ARG RUNTIME_SPARK_MASTER=local[4]
+ARG RUNTIME_DRIVER_CORES=4
+ARG RUNTIME_DRIVER_MEMORY=20g
+ARG RUNTIME_EXECUTOR_CORES=4
+ARG RUNTIME_EXECUTOR_MEMORY=20g
+ARG RUNTIME_EXECUTOR_INSTANCES=1
```

Comment:
Can we remove these environments? @glorysdj @Le-Zheng

Reply:
Yes, you can remove these environments if you don't need.